From 3c1871b84b24f14c3b5c6b99ead40c5aa2a0907e Mon Sep 17 00:00:00 2001
From: Greg Sjaardema <gsjaardema@gmail.com>
Date: Fri, 5 Apr 2024 09:09:46 -0600
Subject: [PATCH] CGNS: Allow more liberal version matching

The main difference between CGNS-3.X and CGNS-4.X is the handling
of polyhedral elements.  If the cgns file does not contain those
elements, then the library should be able to read the file correctly.

We have modified the library version from the original 4.4.X to
be 3.99 so that files written by this version of the CGNS library
(assuming they don't use polyhedral elements) will be readable by
applications that are still using CGNS-3.X

This is a kluge and not officially supported by the CGNS developers.
---
 src/cgnslib.c | 11 ++++++-----
 src/cgnslib.h |  5 +++++
 2 files changed, 11 insertions(+), 5 deletions(-)

diff --git a/src/cgnslib.c b/src/cgnslib.c
index d60daf6..78bd25a 100644
--- a/src/cgnslib.c
+++ b/src/cgnslib.c
@@ -434,12 +434,13 @@ int cg_open(const char *filename, int mode, int *file_number)
         /* This code allows reading version newer than the lib,
                as long as the 1st digit of the versions are equal */
             if ((cg->version / 1000) > (CGNSLibVersion / 1000)) {
-                cgi_error("A more recent version of the CGNS library created the file. Therefore, the CGNS library needs updating before reading the file '%s'.",filename);
-                return CG_ERROR;
-            }
+	      cgi_warning("A more recent version of the CGNS library created the file. Assuming it is OK to read, but may have problems  (%d vs %d).",
+			  cg->version, CGNSLibVersion);
+            }
             /* warn only if different in second digit */
-            if ((cg->version / 100) > (CGNSLibVersion / 100)) {
-                cgi_warning("The file being read is more recent that the CGNS library used");
+	    else if ((cg->version / 100) > (CGNSLibVersion / 100)) {
+	      cgi_warning("The file being read is more recent than the CGNS library used (%d vs %d).",
+			  cg->version, CGNSLibVersion);
             }
         }
 #if CG_SIZEOF_SIZE == 32
diff --git a/src/cgnslib.h b/src/cgnslib.h
index 476c768..a4b66df 100644
--- a/src/cgnslib.h
+++ b/src/cgnslib.h
@@ -43,8 +43,13 @@
 #ifndef CGNSLIB_H
 #define CGNSLIB_H
 
+#if 0
 #define CGNS_VERSION 4400
 #define CGNS_DOTVERS 4.40
+#else
+#define CGNS_VERSION 3990
+#define CGNS_DOTVERS 3.99
+#endif
 
 #define CGNS_COMPATVERSION 2540
 #define CGNS_COMPATDOTVERS 2.54
-- 
2.39.3 (Apple Git-146)

